### PR TITLE
An Example Test on `Check_Systems()`

### PR DESCRIPTION
**Pull request by eliteabhi**
*wooo....this is gonna be a lot...*

---

### TL;DR ~ ***I made an example test for how to generally go about unit testing***
I also had to change...*a lot*

---

**I might repeat a lot of things that I PRd already but I'm very forgetful so bare with me**

## Implimentation
There are a couple of new files that have been added
- ``include/`toolchain.h```
- ```lib/elapsedMillis.h```
- ```test/testing_globals.h```
- ```test/test_example/test_example.h```
- ```test/test_example/test_example_health_check.cpp```

## Extra Notes:
1. I tried to document everything I did in the tests as well as concisely name my definitions
2. I did have to re-write a lot of the pre-existing code to integrate the toolchain changes, any pre-existing code *should* work the same as before
3. In order to run the tests you have to change your environment to ```env:testing```
4. The default environment is ```env:teensy-release``` but I suggest using ```env:teensy-debug``` when building **UNLESS** you're PRing to a ```release/``` branch

<sub> Mention [stepsize] in a comment if you'd like to report some technical debt. See examples [here](https://app.stepsize.com/api/demo-pr-redirect).</sub>

### CODE REVIEW

**Review by juan00juan0 — APPROVED** [review]
Everything looks fine!

**Review by anjusoman — COMMENTED** [review]
Looks good, I think I understood the testing example, will let you know if I have any questions. 

Also noticed this:
functions.h 133-135: should it be r_accel[0] for all three? or r_accel[0], r_accel[1], r_accel[2]?

**Comment by eliteabhi** [comment]
> Looks good, I think I understood the testing example, will let you know if I have any questions. 
> 
> Also noticed this:
> functions.h 133-135: should it be r_accel[0] for all three? or r_accel[0], r_accel[1], r_accel[2]?
> 
> 
> 

*sigh*....yes it's supposed to be that...

Thanks for pointing it out lol

**Comment by eliteabhi** [comment]
> > Looks good, I think I understood the testing example, will let you know if I have any questions. 
> > 
> > Also noticed this:
> > functions.h 133-135: should it be r_accel[0] for all three? or r_accel[0], r_accel[1], r_accel[2]?
> > 
> > 
> > 
> 
> *sigh*....yes it's supposed to be that...
> 
> Thanks for pointing it out lol

I'll fix it and re-merge
